src/main/kotlin/org/rust/lang/core/psi/impl/RustFile.kt

```diff
@@ -32,7 +34,12 @@ class RustFile(
     override val `super`: RustMod?
         get() = RustModulesIndex.getSuperFor(this)
 
-    override val modName: String? = if (name != RustMod.MOD_RS) FileUtil.getNameWithoutExtension(name) else parent?.name
+    override val modName: String? = when {
```

Comment:
I'd rather leave the `modName` of the root crate anonymous and handle extern crates specially in `createLookupElement`. 

In Rust, root modules are anonymous, and it is for a good reason. There is really no good way to identify a particular crate. For example, you can have two crates with the same "names", but different versions. In reality, what happens is that rustc is invoked with a mapping "external_crate_name -> location of the external crate". That is, mapping of a crate name to a particular root module file is *internal* to the crate being compiled, there is no single shared global namespace of crates.